python/make_finemap_inputs.py

```diff
@@ -211,15 +203,44 @@ def generate_bed(sumstats,
         lead_snps.append(lead_snp)
 
     if len(bed_frames) > 0:
-        regions = pd.concat(bed_frames).sort_values(['chrom', 'start'])
+        bed = BedTool.from_dataframe(pd.concat(bed_frames).sort_values(['chrom', 'start']))
+        lead_snps = pd.concat(lead_snps, axis=1).T
         if not no_merge:
+            print(len(sumstats[0].index))
+            print("Window size {}".format(window))
             print("Merging regions. Before")
-            print(regions)
-            regions = merge_regions(regions, args.max_region_width)
+            print(bed)
+            bed = bed.merge().saveas()
+            bed_oversized = bed.filter(lambda x: len(x) > max_region_width).saveas()
+            print("Oversized regions")
+            print(bed_oversized)
+            if (len(bed_oversized)) > 0:
+                # keep in-size regions
+                bed1 = bed.filter(lambda x: len(x) <= max_region_width).saveas()
+                
+                # filter to oversized regions
+                sumstats = map(lambda x: filter_sumstat(x, bed_oversized), sumstats)
```

Comment:
this is unreadable to overwrite sumstats with filtered data... easy enough now but any longer function will get hard to read and error prone. you don't use filtered sumstats anywhere else but in recursion so either don't store the value but pass on the fly to recursion or then change variable name to sumstats_oversize_regions or similar

Reply:
good catch! addressed